src/aligners.rs

```diff
@@ -10,16 +20,217 @@ pub mod layer;
 pub mod nw;
 pub mod nw_lib;
 
+#[derive(Clone)]
+enum Gradient {
+    NoGradient(Color, Color), //(expanded_color,explored color)
+    Gradient(Color, Color),   //(start color, end color)
+    TurboGradient(f32, f32), //(start value, end value); start < end; start > 0 && end > 0; start < 1 && end <= 1
+}
+
+#[derive(Clone)]
+struct ColorScheme {
+    gradient: Gradient,
+    bg_color: Color,
+}
+
+// let default_colors: ColorScheme = ColorScheme{};
+
+#[derive(Clone)]
+pub struct Config {
+    cell_size: usize,
+    prescaler: usize, //for scaling image
+    filepath: String, //maybe &str instead
+    drawing: bool,
+    delay: Cell<f32>,
+    saving: bool,
+    colors: ColorScheme,
+}
+
+impl Default for Config {
+    fn default() -> Self {
+        Self {
+            cell_size: 8,
+            prescaler: 1,
+            saving: false,
+            filepath: String::from(""),
+            drawing: false,
+            delay: Cell::new(0.2),
+            colors: ColorScheme {
+                gradient: Gradient::NoGradient(Color::BLUE, Color::RGB(128, 0, 128)),
+                bg_color: Color::BLACK,
+            },
+        }
+    }
+}
+
+//Saves canvas to bmp file
+pub fn save_canvas(
+    canvas: &sdl2::render::Canvas<sdl2::video::Window>,
+    filepath: &str,
+    number: usize,
+) {
+    let pixel_format = canvas.default_pixel_format();
+    let mut pixels = canvas.read_pixels(canvas.viewport(), pixel_format).unwrap();
+    let (width, height) = canvas.output_size().unwrap();
+    let pitch = pixel_format.byte_size_of_pixels(width as usize);
+    let surf = sdl2::surface::Surface::from_data(
+        pixels.as_mut_slice(),
+        width,
+        height,
+        pitch as u32,
+        pixel_format,
+    )
+    .unwrap();
+    surf.save_bmp(format!("{}{}.bmp", filepath, number))
+        .unwrap_or_else(|error| {
+            print!("Problem saving the file: {:?}", error);
+        });
+}
+
 /// A visualizer can be used to visualize progress of an implementation.
 pub trait Visualizer {
+    fn init(&mut self, config: &Config, len1: u32, len2: u32) {}
```

Comment:
I think init and draw should be methods on the Visualizer struct. Not in the trait